### PR TITLE
Add test for duplicate uploads of a Puppet module

### PR DESCRIPTION
**Pull request by Ichimonji10**
Add a new test case, `DuplicateUploadsTestCase`, in module
`pulp_smash.tests.puppet.api_v2.test_duplicate_uploads`. This module
uploads and imports a Puppet module into a single repository twice in a
row, and verifies that the procedure completes without any issues. The
new test case passes. Tests executed with:

    python -m unittest2 \
        pulp_smash.tests.puppet.api_v2.test_duplicate_uploads

Related to #81.

### CODE REVIEW

**Comment by Ichimonji10** [comment]
This new module is highly similar to `pulp_smash.tsets.rpm.api_v2.test_duplicate_uploads`. I'm leaving the task of inventing a common parent class parent class, mixin, or something else to a future PR.

File: pulp_smash/tests/puppet/api_v2/test_duplicate_uploads.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+
+def _upload_import_puppet_module(server_config, puppet_module, repo_href):
+    """Upload a puppet module and import it into a repository.
+
+    Return the JSON-decoded body of the call report received when importing the
+    Puppet module.
+    """
+    client = api.Client(server_config, api.json_handler)
+    malloc = client.post(CONTENT_UPLOAD_PATH)
```
Comment (elyezer) [inline pulp_smash/tests/puppet/api_v2/test_duplicate_uploads.py]: malloc?

**Comment by elyezer** [comment]
LGTM

My question about malloc is that it does not makes 100% sense to me even though I got the idea of that variable being a placeholder for capturing some specific information about the Puppet Module Upload.

File: pulp_smash/tests/puppet/api_v2/test_duplicate_uploads.py (hunk 1 above)
Comment (Ichimonji10) [inline pulp_smash/tests/puppet/api_v2/test_duplicate_uploads.py]: This is a reference to the C `malloc()` function. When called, `malloc()` allocates the requested amount of memory and returns a pointer to the beginning of that section of memory. The user must later `free()` that memory. A failure to do so results in a memory leak, and a duplicate attempt to free memory results in a segfault or some other critical error.

Pulp's handling of content uploads is reminiscent of `malloc()`. When one makes this HTTP call, the server returns an href (kinda like the C pointer) that must later be used to end the upload request. A failure to do so is a resource leak, and a duplicate attempt results (should result?) in an error.

Comment (Ichimonji10) [inline pulp_smash/tests/puppet/api_v2/test_duplicate_uploads.py]: > C dynamic memory allocation refers to performing manual memory management [via] malloc, realloc, calloc and free.

This is basically my sense of programming humour. :wink:

Comment (elyezer) [inline pulp_smash/tests/puppet/api_v2/test_duplicate_uploads.py]: Got it. :smile:
